### PR TITLE
Fixes defib dropping, splint dropping/storing exploit/oversight.

### PR DESCRIPTION
**Pull request by somepersondunno**
# About the pull request

Fixes Issue #1315, as well as a common thing I see of medics starting the defib timer, then dropping their defib in order to do other things (pill, suture the patient, etc).

# Explain why it's good for the game

This fixes some nobrainer issues. If this causes any issues, please let me know, as I am very new to this whole thing.

# Testing Photographs and Procedure
<summary>I tested this on a private server and did everything imagineable to interrupt the actions (splinting and defibbing). I splinted myself, moving, dropping and storing the item, and each time the action was interrupted. I did the same to someone else. I also killed then defibbed a mob, and every time I tried something similar the action was interrupted, so this is working as intended.</summary>

# Changelog

:cl:
fix: fixed issues/exploits surrounding defibrillators and splints
/:cl:

### CODE REVIEW

**Comment by carlarctg** [comment]
does this mean swapping hands will break the action? because that would suck ass

**Comment by somepersondunno** [comment]
> does this mean swapping hands will break the action? because that would suck ass

this is an unavoidable issue without programming in another interrupt, but forcing players to wait on tasks like attaching a gun, splinting or defibbing is just part of the game.

**Review by morrowwolf — CHANGES_REQUESTED** [review]
Dropping defib and having it continue is intentional. Please put it back.

**Comment by somepersondunno** [comment]
Fixed.

File: code/game/objects/items/devices/defibrillator.dm [hunk 1]
```diff
@@ -167,7 +167,7 @@
 	playsound(get_turf(src),'sound/items/defib_charge.ogg', 25, 0) //Do NOT vary this tune, it needs to be precisely 7 seconds
 
 	//Taking square root not to make defibs too fast...
-	if(!do_after(user, 7 SECONDS * user.get_skill_duration_multiplier(SKILL_MEDICAL), INTERRUPT_NO_NEEDHAND|BEHAVIOR_IMMOBILE, BUSY_ICON_FRIENDLY, H, INTERRUPT_MOVED, BUSY_ICON_MEDICAL))
+	if(do_after(user, time_to_take * user.get_skill_duration_multiplier(SKILL_MEDICAL), INTERRUPT_NO_NEEDHAND, BUSY_ICON_FRIENDLY, target, INTERRUPT_MOVED, BUSY_ICON_MEDICAL))
```
Comment (harryob) [inline code/game/objects/items/devices/defibrillator.dm:170]: you need to define time_to_take and redefine H to target in definition

**Review by harryob — CHANGES_REQUESTED** [review]

**Comment by Segrain** [comment]
> Dropping defib and having it continue is intentional. Please put it back.

No, it is absolutely _not_.

**Comment by Segrain** [comment]
> > does this mean swapping hands will break the action? because that would suck ass
> 
> this is an unavoidable issue without programming in another interrupt, but forcing players to wait on tasks like attaching a gun, splinting or defibbing is just part of the game. edit: virtually every action that works similarly (on the same type of timer) requires you to wait.

No. Just no. You are trying to remove the entire mechanic that exists for a reason and purpose. Justifying it as fixing a bug is, as the expression goes, throwing out the baby with the water. I am going to save you the trouble of fixing everything else other reviewers have pointed out and simply close this on design principles.
(And yes, I will get to actually writing a proper fix. Soon. It is a sizeable work to go over every usage case of `do_after`'s interrupts and ensure that only those that need be affected are affected.)

**Pull request closed without merging**